### PR TITLE
EigenvectorCentrality now handles orphan nodes and return 0 for isolated nodes

### PR DESCRIPTION
**Pull request by burqen**
issue #4847

### CODE REVIEW

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java [hunk 1]
```diff
@@ -95,6 +96,14 @@ public EigenvectorCentralityArnoldi( Direction relationDirection,
      */
     public Double getCentrality( Node node )
     {
+        if ( !nodeSet.contains( node ) )
+        {
+            return null;
+        }
+        if ( !node.hasRelationship() )
+        {
+            return 0d;
```
Comment (MishaDemianenko) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java]: Lets introduce a well named constant?

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java [hunk 2]
```diff
@@ -191,9 +200,22 @@ public int runIterations( int maxNrIterations )
...
+        {
+            double value = values.get( iter.next() );
+            if ( value < -1E-8 || value > 1E-8 )
+            {
+                firstNoneZeroValue = value;
+                break;
+            }
+        }
+
         // If the first value is negative (possibly the whole vector), negate
```
Comment (MishaDemianenko) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java]: Comment is out of sync with new implementation

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java [hunk 3]
```diff
@@ -191,9 +200,22 @@ public int runIterations( int maxNrIterations )
                 break;
             }
         }
+
+        double firstNoneZeroValue = 0;
+        Iterator<Node> iter = nodeSet.iterator();
+        while ( iter.hasNext() )
+        {
+            double value = values.get( iter.next() );
+            if ( value < -1E-8 || value > 1E-8 )
```
Comment (MishaDemianenko) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java]: looks like `NoneStrictMath.EPSILON` and new negative EPSILON?

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java (hunk 1 above)
Comment (burqen) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java]: This zero is not a flag for anything, it is the actual mathematical centrality value of a node without relationships. Still propose a constant?

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java (hunk 3 above)
Comment (burqen) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java]: Yep, does not exist in 2.2 though

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityPower.java [hunk 4]
```diff
@@ -90,6 +91,14 @@ public EigenvectorCentralityPower( Direction relationDirection,
      */
     public Double getCentrality( Node node )
     {
+        if ( !nodeSet.contains( node ) )
```
Comment (MishaDemianenko) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityPower.java]: Almost all changes in `EigenvectorCentralityPower` are exactly the same as for `EigenvectorCentralityArnoldi`. Looks like we can extract superclass for those 2?

File: community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java (hunk 1 above)
Comment (MishaDemianenko) [inline community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/centrality/EigenvectorCentralityArnoldi.java]: exactly, it's not a flag it's a centrality of detached vertex. so would it be better to have DETACHED_VERTEX_CENTRALITY or smt similar?

**Comment by MishaDemianenko** [comment]
Looks fine for me for 2.2. For 2.3 would propose to made small update after that will include our `NonStrictMath` thingy.

**Pull request merged**